### PR TITLE
Fix #15: Enable OpenTelemetry

### LINKED ISSUE #15: Integration with Quarkus OpenTelemetry
https://github.com/temporalio/sdk-java/tree/master/temporal-opentracing

### PR DESCRIPTION
**Pull request by melloware**
Fix #15: Enable OpenTelemetry

### CODE REVIEW

**Comment by melloware** [comment]
I followed the Temporal examples and docs basically we have to add Interceptors to WorkerClient and WorkerFactory.

Then we need to use th OpenTelementry Shim to forward OpenTracing to OpenTelemetry but luckly i was able to grab an instance of OpenTelemetry from the Quarkus Extension using CDI.

```java
OpenTelemetry openTelemetry = CDI.current().select(OpenTelemetry.class).get();
```

File: pom.xml [hunk 1]
```diff
@@ -52,11 +52,21 @@
...
+            </dependency>
             <dependency>
                 <groupId>io.temporal</groupId>
                 <artifactId>temporal-testing</artifactId>
                 <version>${temporal.version}</version>
             </dependency>
+            <dependency>
+                <groupId>io.opentelemetry</groupId>
+                <artifactId>opentelemetry-opentracing-shim</artifactId>
+                <version>1.32.0</version>
```
Comment (melloware) [inline pom.xml:68]: I am asking Bruno a question on Zulip on how I can access the opentelemetry version so it stays in sync with Quarkus's version.

**Comment by rmanibus** [comment]
I am wondering if we should not use the capabilities registry to check if otel is present instead of forcing the dependency in the project

**Comment by melloware** [comment]
Great idea let me try that.  I was going to ask how to do "optional" dependency but this looks like the correct way.  Will update PR.

**Comment by melloware** [comment]
OK updated to check OpenTelemetry capabilties similar to how the Agroal extension does it.

This allowed me to set the dependency as `<optional>true</optional>`

**Comment by rmanibus** [comment]
looks awesome ! I am just wondering, are the other extensions exposing a switch for otel, or should we enable it de-facto if the extension is present ?

File: extension/runtime/src/main/java/io/quarkiverse/temporal/WorkerFactoryRecorder.java [hunk 2]
```diff
@@ -78,12 +86,23 @@ public void createWorker(String name, List<Class<?>> workflows,
...
         workerFactory.start();
+
+        // bridge Temporal OpenTracing to OpenTelemetry
+        if (buildtimeConfig.enableTelemetry()) {
+            OpenTelemetry openTelemetry = CDI.current().select(OpenTelemetry.class).get();
+            if (openTelemetry == null) {
+                throw new IllegalStateException("OpenTelemetry not available");
+            }
+
+            io.opentracing.Tracer tracer = OpenTracingShim.createTracerShim(openTelemetry);
```
Comment (rmanibus) [inline extension/runtime/src/main/java/io/quarkiverse/temporal/WorkerFactoryRecorder.java:102]: how is the tracer added to the interceptor ?

Comment (rmanibus) [reply extension/runtime/src/main/java/io/quarkiverse/temporal/WorkerFactoryRecorder.java:102]: is it relying on a global singleton somewhere ?

Comment (rmanibus) [reply extension/runtime/src/main/java/io/quarkiverse/temporal/WorkerFactoryRecorder.java:102]: in this readme: https://github.com/temporalio/sdk-java/tree/master/temporal-opentracing

I see that the shim is added to the global context like this:

```
    io.opentracing.Tracer tracer = OpenTracingShim.createTracerShim(openTelemetry);
    GlobalTracer.registerIfAbsent(tracer);
```

Comment (rmanibus) [reply extension/runtime/src/main/java/io/quarkiverse/temporal/WorkerFactoryRecorder.java:103]: have you been able to confirm that it is actually working ?

**Review by rmanibus — APPROVED** [review]

File: extension/runtime/src/main/java/io/quarkiverse/temporal/WorkerFactoryRecorder.java (hunk 2 above)
Comment (melloware) [reply extension/runtime/src/main/java/io/quarkiverse/temporal/WorkerFactoryRecorder.java:103]: Yes when you even click on the Dev Ui car workers i can. See temporal connect to OpenTelemetry.

**Pull request merged**